lib/spack/spack/binary_distribution.py

```diff
@@ -1399,6 +1399,16 @@ def _build_tarball_in_stage_dir(spec: Spec, out_url: str, stage_dir: str, option
 
     checksum, _ = _do_create_tarball(tarfile_path, binaries_dir, buildinfo)
 
+    if spec_file.endswith(".yaml"):
```

Comment:
Why this and not

```python
        elif spec_file.endswith(".yaml"):
            spec_dict = syaml.load(content)
```

a few lines down?

Reply:
Or, without reinventing the wheel:

```python
   spec_dict = Spec.from_specfile(spec_file).to_dict()
    spec_dict["buildcache_layout_version"] = CURRENT_BUILD_CACHE_LAYOUT_VERSION
    spec_dict["binary_cache_checksum"] = {"hash_algorithm": "sha256", "hash": checksum}
 ```

Reply:
Or even #41321

Reply:
I assumed there was some reason the .json file format was required for the buildcache creation.

Reply:
Also, it makes the packages more consistent; new packages are created wtih spec.json files; only old Spack versions made spec.yaml files, so eventually the support for yaml files here may go away; converting them makes them more future-proof.

Reply:
I remember being told by Todd a few years ago that YAML files were significantly more expensive to parse than JSON files—I don't know if that's still the case.